hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/recon/TestReconScmHASnapshot.java

```diff
@@ -0,0 +1,136 @@
+    conf.setBoolean(OZONE_SCM_HA_ENABLE_KEY, true);
+    conf.setBoolean(
+        ReconServerConfigKeys.OZONE_RECON_SCM_SNAPSHOT_ENABLED, true);
+    conf.setInt(ReconServerConfigKeys.OZONE_RECON_SCM_CONTAINER_THRESHOLD, 0);
+    conf.setInt(ScmConfigKeys.OZONE_DATANODE_PIPELINE_LIMIT, 5);
+    cluster = MiniOzoneCluster.newBuilder(conf)
+        .setNumDatanodes(4)
+        .includeRecon(true)
+        .build();
+    cluster.waitForClusterToBeReady();
+  }
+
+  @Test
+  public void testScmHASnapshot() throws Exception {
+    GenericTestUtils.LogCapturer logCapturer = GenericTestUtils.LogCapturer
+        .captureLogs(LoggerFactory.getLogger(
+            ReconStorageContainerManagerFacade.class));
+
+    List<ContainerInfo> reconContainers = cluster.getReconServer()
+        .getReconStorageContainerManager().getContainerManager()
+        .getContainers();
+    assertEquals(0, reconContainers.size());
+
+    ReconNodeManager nodeManager;
+    nodeManager = (ReconNodeManager) cluster.getReconServer()
+        .getReconStorageContainerManager().getScmNodeManager();
+    long keyCountBefore = nodeManager.getNodeDBKeyCount();
+
+    //Stopping Recon to add Containers in SCM
+    cluster.stopRecon();
+
+    ContainerManager containerManager;
+    containerManager = cluster.getStorageContainerManager()
+        .getContainerManager();
+
+    for (int i = 0; i < 10; i++) {
+      containerManager.allocateContainer(new RatisReplicationConfig(
+          HddsProtos.ReplicationFactor.ONE), "testOwner");
+    }
+
+    cluster.startRecon();
+
+    //ContainerCount after Recon DB is updated with SCM DB
+    containerManager = cluster.getStorageContainerManager()
+        .getContainerManager();
+    ContainerManager reconContainerManager = cluster.getReconServer()
+        .getReconStorageContainerManager().getContainerManager();
+    assertTrue(logCapturer.getOutput()
+        .contains("Downloaded SCM Snapshot from Leader SCM"));
+    assertTrue(logCapturer.getOutput()
+        .contains("Recon Container Count: " + reconContainers.size() +
+        ", SCM Container Count: " + containerManager.getContainers().size()));
+    assertEquals(containerManager.getContainers().size(),
+        reconContainerManager.getContainers().size());
+
+    //PipelineCount after Recon DB is updated with SCM DB
+    PipelineManager scmPipelineManager = cluster.getStorageContainerManager()
+        .getPipelineManager();
+    PipelineManager reconPipelineManager = cluster.getReconServer()
+        .getReconStorageContainerManager().getPipelineManager();
+    assertEquals(scmPipelineManager.getPipelines().size(),
+        reconPipelineManager.getPipelines().size());
+
+    //NodeCount after Recon DB updated with SCM DB
+    nodeManager = (ReconNodeManager) cluster.getReconServer()
+        .getReconStorageContainerManager().getScmNodeManager();
+    long keyCountAfter = nodeManager.getNodeDBKeyCount();
+    assertEquals(keyCountAfter, keyCountBefore);
```

Comment:
Can we extract this into a common assertion function shared between the HA and non HA variants of this test?